### DISCUSSION TITLE
Ask for help: Add image into the chatInputArea

### DISCUSSION THREAD

**Discussion by CSLukkun** [ant-design/pro-chat#71] [Q&A]
Thanks for your attention.

I want to ask how to add images or other files to the chat input area.

Anyone can provide me with some guidance to deal with it?

Thanks again

**Comment by ONLY-yours** [comment]
nice question, we are considering the scope of opening, should we default open it in chatInputArea ? 

i thinks maybe its not a good idea ,if we default open it in chatInputArea, something may not so flexible

- how to combine message and image 
- should we defalut use base 64? but some one may be like to use cdn link
- if more files we should supprt , such as pdf、docs ,how this work?

now we reserved an api  named renderInput （u can find it in source code, but not actually finished it）

now this api is work such as this
```js
 <div ref={areaHtml}>{renderInput ?? <ChatInputArea />}</div>
```
but actually will have some args, we still design about it, such as this

```ts
renderInput: {
   defaultDom: ReactNode,
   onSend,
   onDeleteAll
   // TODO
}
```

**Reply by CSLukkun** [reply]
Yes, I found that. But I thought It is too tightly coupled with the store layer of prochat.

So, I just wonder how to design in the next.

Thanks again.

**Comment by ONLY-yours** [comment]
now u can use Hooks and renderInput to achieve it 

- sendMessage can allowed u to seed custom message

![image](https://github.com/ant-design/pro-chat/assets/52664827/8bb4eb77-bec9-4cf0-bac0-839a83ebdf49)

than in action render, u can put a antd picture upload components,than set a state in it

```ts
const [files,setUploadFiles] = useState()

    <div style={{ background: theme.colorBgLayout }}>
      <ProChat
        actions={{
          render: (defaultDoms) => {
                return <Upload  onChange={setUploadFiles} /> // this is antd upload
            }
        }}
        renderInput={()=>{
            return <CustomComponents/>
        }}
      />
    </div>

```

in CustomComponents U can use ProChat.sendMessage(message + files) to send, and u can choose the way to deal with files than send it

we will provider a simpler way and docs deme to help u in next few weeks

**Reply by CSLukkun** [reply]
Thanks for your consideration. It makes sense to me.

**Comment by ONLY-yours** [comment]
@CSLukkun look about this ：https://github.com/ant-design/pro-chat/pull/74
